slick-testkit/src/codegen/scala/slick/test/codegen/GenerateRoundtripSources.scala

```diff
@@ -29,15 +29,21 @@ object GenerateRoundtripSources {
     })
     val a2 = profile.createModel(ignoreInvalidDefaults=false).map(m => new SourceCodeGenerator(m) {
       override def Table = new Table(_){
-        override def hlistEnabled = true
         override def autoIncLastAsOption = true
+        override def hlistEnabled = columns.size > 22
+      }
+    })
+    val a3 = profile.createModel(ignoreInvalidDefaults=false).map(m => new SourceCodeGenerator(m) {
+      override def Table = new Table(_){
+        override def hlistEnabled = true
       }
     })
     val db = Database.forURL(url=url, driver=jdbcDriver, keepAliveConnection=true)
-    val (gen,gen2) = try Await.result(db.run(ddl.create >> (a1 zip a2)), Duration.Inf) finally db.close
+    val ((gen,gen2),gen3) = try Await.result(db.run(ddl.create >> ((a1 zip a2) zip a3)), Duration.Inf) finally db.close
     val pkg = "slick.test.codegen.roundtrip"
     gen.writeToFile( "slick.jdbc.H2Profile", args(0), pkg )
     gen2.writeToFile( "slick.jdbc.H2Profile", args(0), pkg+"2" )
+    gen3.writeToFile( "slick.jdbc.H2Profile", args(0), pkg+"3" )
```

Comment:
this generates the file below. Not sure that file compiles. We'll have to see after the above compilation problems are fixed and then see what needs to be done.